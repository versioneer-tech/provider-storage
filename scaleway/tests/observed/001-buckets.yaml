---
apiVersion: object.scaleway.upbound.io/v1alpha1
kind: Bucket
metadata:
  annotations:
    crossplane.io/composition-resource-name: bucket-alice-001
  name: alice-001-alice-001
spec:
  forProvider:
    name: alice-001-alice-001-abcde-fghij
status:
  conditions:
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: Available
    status: "True"
    type: Ready
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: ReconcileSuccess
    status: "True"
    type: Synced
---
apiVersion: object.scaleway.upbound.io/v1alpha1
kind: Bucket
metadata:
  annotations:
    crossplane.io/composition-resource-name: bucket-alice-shared-001
  name: alice-001-alice-shared-001
spec:
  forProvider:
    name: alice-001-alice-shared-001-abcde-fghij
status:
  conditions:
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: Available
    status: "True"
    type: Ready
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: ReconcileSuccess
    status: "True"
    type: Synced
