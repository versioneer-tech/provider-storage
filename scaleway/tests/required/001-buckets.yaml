---
apiVersion: v1
data:
  credentials: ewogICJhY2Nlc3Nfa2V5IjogImFjY2Vzc19rZXkiLAogICJzZWNyZXRfa2V5IjogInNlY3JldF9rZXkiLAogICJvcmdhbml6YXRpb25faWQiOiAib3JnYW5pemF0aW9uX2lkIiwKICAidXNlcl9pZCI6ICJ1c2VyX2lkIiwKICAicmVnaW9uIjogImZyLXBhciIsCiAgInpvbmUiOiAiZnItcGFyLTEiCn0K
kind: Secret
metadata:
  name: storage-scaleway
  namespace: crossplane-system
type: Opaque
---
apiVersion: v1
data:
  random-string: abcde-fghij
kind: ConfigMap
metadata:
  name: random-string
  namespace: crossplane-system
---
apiVersion: iam.scaleway.upbound.io/v1alpha1
kind: Application
metadata:
  annotations:
    crossplane.io/composition-resource-name: application-alice-001
  name: alice-001
spec: {}
status:
  atProvider:
    id: application_id_alice-001
  conditions:
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: Available
    status: "True"
    type: Ready
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: ReconcileSuccess
    status: "True"
    type: Synced

