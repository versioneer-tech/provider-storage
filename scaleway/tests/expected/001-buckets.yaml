---
apiVersion: pkg.internal/v1beta1
kind: XStorage
metadata:
  labels:
    crossplane.io/claim-name: alice-001
    crossplane.io/claim-namespace: default
  name: alice-001
spec:
  buckets:
  - bucketName: alice-001
    discoverable: false
  - bucketName: alice-shared-001
    discoverable: true
  owner: alice-001
status:
  conditions:
  - lastTransitionTime: "2024-01-01T00:00:00Z"
    message: 'Unready resources: api-key-alice-001, application-alice-001, bucket-policy-alice-001,
      and 2 more'
    reason: Creating
    status: "False"
    type: Ready
---
apiVersion: iam.scaleway.upbound.io/v1alpha1
kind: ApiKey
metadata:
  annotations:
    crossplane.io/composition-resource-name: api-key-alice-001
  labels:
    crossplane.io/composite: alice-001
  name: alice-001
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: XStorage
    name: alice-001
    uid: ""
spec:
  forProvider:
    applicationIdRef:
      name: alice-001
  providerConfigRef:
    name: storage-scaleway
  writeConnectionSecretToRef:
    name: alice-001
    namespace: default
---
apiVersion: iam.scaleway.upbound.io/v1alpha1
kind: Application
metadata:
  annotations:
    crossplane.io/composition-resource-name: application-alice-001
  labels:
    crossplane.io/composite: alice-001
  name: alice-001
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: XStorage
    name: alice-001
    uid: ""
spec:
  forProvider:
    name: alice-001
  providerConfigRef:
    name: storage-scaleway
---
apiVersion: object.scaleway.upbound.io/v1alpha1
kind: Bucket
metadata:
  annotations:
    crossplane.io/composition-resource-name: bucket-alice-001-alice-001
    xstorages.pkg.internal/bucket-name: alice-001
    xstorages.pkg.internal/discoverable: "False"
    xstorages.pkg.internal/owner: alice-001
  labels:
    crossplane.io/composite: alice-001
  name: alice-001-alice-001
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: XStorage
    name: alice-001
    uid: ""
spec:
  forProvider:
    name: alice-001-alice-001-abcde-fghij
  providerConfigRef:
    name: storage-scaleway
---
apiVersion: object.scaleway.upbound.io/v1alpha1
kind: Bucket
metadata:
  annotations:
    crossplane.io/composition-resource-name: bucket-alice-001-alice-shared-001
    xstorages.pkg.internal/bucket-name: alice-shared-001
    xstorages.pkg.internal/discoverable: "True"
    xstorages.pkg.internal/owner: alice-001
  labels:
    crossplane.io/composite: alice-001
  name: alice-001-alice-shared-001
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: XStorage
    name: alice-001
    uid: ""
spec:
  forProvider:
    name: alice-001-alice-shared-001-abcde-fghij
  providerConfigRef:
    name: storage-scaleway
---
apiVersion: object.scaleway.upbound.io/v1alpha1
kind: Policy
metadata:
  annotations:
    crossplane.io/composition-resource-name: bucket-policy-alice-001
  labels:
    crossplane.io/composite: alice-001
  name: alice-001
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: XStorage
    name: alice-001
    uid: ""
spec:
  forProvider:
    bucket: alice-001-alice-001-abcde-fghij
    policy: |-
      {
        "Version": "2023-04-17",
        "Statement": [
          {
            "Effect": "Allow",
            "Principal": {
              "SCW": [
                "user_id:user_id",
                "application_id:application_id_alice-001"
              ]
            },
            "Action": [
              "s3:*"
            ],
            "Resource": [
              "alice-001-alice-001-abcde-fghij",
              "alice-001-alice-001-abcde-fghij/*"
            ]
          },
          {
            "Effect": "Allow",
            "Principal": {
              "SCW": []
            },
            "Action": [
              "s3:ListBucket",
              "s3:GetObject",
              "s3:PutObject",
              "s3:DeleteObject"
            ],
            "Resource": [
              "alice-001-alice-001-abcde-fghij",
              "alice-001-alice-001-abcde-fghij/*"
            ]
          },
          {
            "Effect": "Allow",
            "Principal": {
              "SCW": []
            },
            "Action": [
              "s3:ListBucket",
              "s3:GetObject"
            ],
            "Resource": [
              "alice-001-alice-001-abcde-fghij",
              "alice-001-alice-001-abcde-fghij/*"
            ]
          }
        ]
      }
  providerConfigRef:
    name: storage-scaleway
---
apiVersion: object.scaleway.upbound.io/v1alpha1
kind: Policy
metadata:
  annotations:
    crossplane.io/composition-resource-name: bucket-policy-alice-shared-001
  labels:
    crossplane.io/composite: alice-001
  name: alice-shared-001
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: XStorage
    name: alice-001
    uid: ""
spec:
  forProvider:
    bucket: alice-001-alice-shared-001-abcde-fghij
    policy: |-
      {
        "Version": "2023-04-17",
        "Statement": [
          {
            "Effect": "Allow",
            "Principal": {
              "SCW": [
                "user_id:user_id",
                "application_id:application_id_alice-001"
              ]
            },
            "Action": [
              "s3:*"
            ],
            "Resource": [
              "alice-001-alice-shared-001-abcde-fghij",
              "alice-001-alice-shared-001-abcde-fghij/*"
            ]
          },
          {
            "Effect": "Allow",
            "Principal": {
              "SCW": []
            },
            "Action": [
              "s3:ListBucket",
              "s3:GetObject",
              "s3:PutObject",
              "s3:DeleteObject"
            ],
            "Resource": [
              "alice-001-alice-shared-001-abcde-fghij",
              "alice-001-alice-shared-001-abcde-fghij/*"
            ]
          },
          {
            "Effect": "Allow",
            "Principal": {
              "SCW": []
            },
            "Action": [
              "s3:ListBucket",
              "s3:GetObject"
            ],
            "Resource": [
              "alice-001-alice-shared-001-abcde-fghij",
              "alice-001-alice-shared-001-abcde-fghij/*"
            ]
          }
        ]
      }
  providerConfigRef:
    name: storage-scaleway
---
apiVersion: iam.scaleway.upbound.io/v1alpha1
kind: Policy
metadata:
  annotations:
    crossplane.io/composition-resource-name: iam-policy-alice-001
  labels:
    crossplane.io/composite: alice-001
  name: alice-001
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: XStorage
    name: alice-001
    uid: ""
spec:
  forProvider:
    applicationIdRef:
      name: alice-001
    name: object-storage-full-access-alice-001
    rule:
    - organizationId: organization_id
      permissionSetNames:
      - ObjectStorageFullAccess
  providerConfigRef:
    name: storage-scaleway
